app/org/sagebionetworks/bridge/dynamodb/DynamoUploadFieldDefinition.java

```diff
@@ -0,0 +1,90 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import javax.annotation.Nonnull;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.models.upload.UploadFieldDefinition;
+import org.sagebionetworks.bridge.models.upload.UploadFieldType;
+import org.sagebionetworks.bridge.validators.Validate;
+
+/**
+ * Dynamo DB implementation of UploadFieldDefinition. While there is nothing specific to Dynamo DB in this class, this
+ * class exists to distinguish itself from potential other implementations.
+ */
+@JsonDeserialize(builder = DynamoUploadFieldDefinition.Builder.class)
+public class DynamoUploadFieldDefinition implements UploadFieldDefinition {
+    private final @Nonnull String name;
```

Comment:
Is @Nonnull a new feature in Java 8?  Not sure if we have fully migrated to Java 8 yet.